ci/kind/kind-setup.sh

```diff
@@ -244,25 +245,68 @@ function configure_extra_networks {
   done
 }
 
+function get_bridge_id {
+  local bridge_id=$(docker network inspect kind -f {{.ID}})
+  local return_value=$?
+  if [ $return_value -eq 0 ]; then
+        echo "$bridge_id"
+    else
+        echo "Unable to get kind bridge id, Docker command failed with exit code $return_value"
+        exit 1
+    fi
+}
+
+# add_kind_ipam_routes  add routes for non-ipam test-pods.
+function add_kind_ipam_routes {
+     kubectl get nodes -o wide --no-headers=true |  awk '{print $1}' | while read nodes; do
+      local pod_cidr=$(kubectl get nodes "${nodes}" -o yaml | grep podCIDR | awk 'FNR == 1 {print$2}')
+      local node_ip=$(kubectl get nodes "${nodes}" -o wide | awk 'FNR == 2 {print$6}')
+      sudo ip route add "$pod_cidr" via "$node_ip"
+     done
+}
+
+# delete_kind_ipam_routes delete the routes of non-ipam test-pods from the Kind testbed.
+function delete_kind_ipam_routes { 
+    cluster=$(kind get clusters)
+    if [ ! -z "$cluster" ]; then
+        cidrs=$(kubectl get nodes -o jsonpath='{.items[*].spec.podCIDR}')
+          echo -e "$cidrs" | tr ' ' '\n'| while read cidr; do
+                del_routes=$(ip route | awk -v cidr="$cidr" '$0 ~ cidr {print$1}')
+                if [[ -z $del_routes ]]; then
+                    echo "Test pods routes are not available on testbed"
+                    exit 0
+                else
+                    sudo ip route del $del_routes || true
+                fi 
+            done
+    fi
+}
+
 function configure_vlan_subnets {
   if [[ -z $VLAN_SUBNETS || -z $VLAN_ID ]]; then
     return
   fi
   echo "Configuring VLAN subnets"
 
-  bridge_id=$(docker network inspect kind -f {{.ID}})
-  bridge_interface="br-${bridge_id:0:12}"
-  vlan_interface="br-${bridge_id:0:7}.$VLAN_ID"
-
-  docker_run_with_host_net ip link add link $bridge_interface name $vlan_interface type vlan id $VLAN_ID
-  docker_run_with_host_net ip link set $vlan_interface up
   IFS=',' read -r -a vlan_subnets <<< "$VLAN_SUBNETS"
-  for s in "${vlan_subnets[@]}" ; do
+  IFS=',' read -r -a vlan_ids <<< "$VLAN_ID"
+  i=0
+
+  for s in "${vlan_subnets[@]}"; do
+    vlan_id=${vlan_ids[$i]}
```

Comment:
The original parameter supports one vlan shared with multiple subnets, and there are test cases still passing parameters in that way. You should use two different IFS to differentiate subnets using the same vlan and subnets using different vlans, otherwise this change would break other tests.

Reply:
changed it, could you confirm once?

Reply:
It's not what I meant. I mean the current code would break the following usage: https://github.com/antrea-io/antrea/blob/943d9b3df18e301e7b42199f671296576d06aa2e/ci/kind/test-e2e-kind.sh#L296-L302

You need a way to express two subnets used by a single VLAN and two subnets used by two VLANs differently.

Reply:
done, I used flag to differentiate that. if single_vlan flag is true then it means subnets have one vlan and else multiple vlan.